src/dataflow/source/file.rs

```diff
@@ -181,21 +176,93 @@ fn read_file_task<Ctor, I, Out, Err>(
                 _h: handle,
             };
 
-            let iter = match iter_ctor(Box::new(file)) {
-                Ok(iter) => iter,
-                Err(err) => {
-                    error!("Failed to create tailed file source: {}", err);
-                    return;
+            iter_ctor(Box::new(file))
+        }
+    };
+
+    match iter {
+        Ok(i) => send_records(i, tx, activator),
+        Err(e) => error!("Failed to obtain records from file. Error {}", e),
+    };
+}
+
+/// Timestamp history map is of format [pid1: (ts1, offset1), (ts2, offset2), pid2: (ts1, offset)...].
```

Comment:
not quite accurate, the tuples are (partition_count, ts, offset)

Reply:
You fixed this in your next PR right?